### ISSUE TITLE
tf error on import

### ISSUE THREAD

**Issue by StuartIanNaylor** [Qengineering/TensorFlow-Raspberry-Pi_64-bit#3]
installed as with https://qengineering.eu/install-tensorflow-2.4.0-on-raspberry-64-os.html pi os 64 

```
Traceback (most recent call last):
  File "configure.py", line 24, in <module>
    import tensorflow as tf
  File "/home/pi/google-kws/venv/lib/python3.7/site-packages/tensorflow/__init__.py", line 41, in <module>
    from tensorflow.python.tools import module_util as _module_util
  File "/home/pi/google-kws/venv/lib/python3.7/site-packages/tensorflow/python/__init__.py", line 48, in <module>
    from tensorflow.python import keras
  File "/home/pi/google-kws/venv/lib/python3.7/site-packages/tensorflow/python/keras/__init__.py", line 27, in <module>
    from tensorflow.python.keras import models
  File "/home/pi/google-kws/venv/lib/python3.7/site-packages/tensorflow/python/keras/models.py", line 26, in <module>
    from tensorflow.python.keras.engine import functional
  File "/home/pi/google-kws/venv/lib/python3.7/site-packages/tensorflow/python/keras/engine/functional.py", line 38, in <module>
    from tensorflow.python.keras.engine import training as training_lib
  File "/home/pi/google-kws/venv/lib/python3.7/site-packages/tensorflow/python/keras/engine/training.py", line 56, in <module>
    from tensorflow.python.keras.saving import hdf5_format
  File "/home/pi/google-kws/venv/lib/python3.7/site-packages/tensorflow/python/keras/saving/hdf5_format.py", line 42, in <module>
    import h5py
  File "/home/pi/google-kws/venv/lib/python3.7/site-packages/h5py/__init__.py", line 46, in <module>
    from ._conv import register_converters as _register_converters
  File "h5py/_conv.pyx", line 1, in init h5py._conv
  File "h5py/h5t.pyx", line 1, in init h5py.h5t
ValueError: numpy.ndarray size changed, may indicate binary incompatibility. Expected 88 from C header, got 80 from PyObject

```
tried to force  reinstall but still the same.

```
(venv) pi@raspberrypi:~/google-kws $ pip install --upgrade --force-reinstall tensorflow-2.4.1-cp37-cp37m-linux_aarch64.whl
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Processing ./tensorflow-2.4.1-cp37-cp37m-linux_aarch64.whl
Collecting opt-einsum~=3.3.0
  Using cached https://www.piwheels.org/simple/opt-einsum/opt_einsum-3.3.0-py3-none-any.whl (65 kB)
Collecting grpcio~=1.32.0
  Using cached grpcio-1.32.0-cp37-cp37m-linux_aarch64.whl
Collecting flatbuffers~=1.12.0
  Using cached https://www.piwheels.org/simple/flatbuffers/flatbuffers-1.12-py2.py3-none-any.whl (15 kB)
Collecting wheel~=0.35
  Using cached https://www.piwheels.org/simple/wheel/wheel-0.36.2-py2.py3-none-any.whl (35 kB)
Collecting google-pasta~=0.2
  Using cached https://www.piwheels.org/simple/google-pasta/google_pasta-0.2.0-py3-none-any.whl (57 kB)
Collecting h5py~=2.10.0
  Using cached h5py-2.10.0-cp37-cp37m-linux_aarch64.whl
Collecting wrapt~=1.12.1
  Using cached wrapt-1.12.1-cp37-cp37m-linux_aarch64.whl
Collecting numpy~=1.19.2
  Using cached numpy-1.19.5-cp37-cp37m-manylinux2014_aarch64.whl (12.4 MB)
Collecting termcolor~=1.1.0
  Using cached https://www.piwheels.org/simple/termcolor/termcolor-1.1.0-py3-none-any.whl (4.8 kB)
Collecting gast==0.3.3
  Using cached https://www.piwheels.org/simple/gast/gast-0.3.3-py2.py3-none-any.whl (9.7 kB)
Collecting tensorboard~=2.4
  Using cached tensorboard-2.4.1-py3-none-any.whl (10.6 MB)
Collecting six~=1.15.0
  Using cached https://www.piwheels.org/simple/six/six-1.15.0-py2.py3-none-any.whl (10 kB)
Collecting keras-preprocessing~=1.1.2
  Using cached https://www.piwheels.org/simple/keras-preprocessing/Keras_Preprocessing-1.1.2-py2.py3-none-any.whl (42 kB)
Collecting tensorflow-estimator<2.5.0,>=2.4.0
  Using cached tensorflow_estimator-2.4.0-py2.py3-none-any.whl (462 kB)
Collecting absl-py~=0.10
  Using cached https://www.piwheels.org/simple/absl-py/absl_py-0.11.0-py3-none-any.whl (127 kB)
Collecting typing-extensions~=3.7.4
  Using cached https://www.piwheels.org/simple/typing-extensions/typing_extensions-3.7.4.3-py3-none-any.whl (22 kB)
Collecting astunparse~=1.6.3
  Using cached https://www.piwheels.org/simple/astunparse/astunparse-1.6.3-py2.py3-none-any.whl (12 kB)
Collecting protobuf>=3.9.2
  Using cached protobuf-3.15.5-py2.py3-none-any.whl (173 kB)
Collecting werkzeug>=0.11.15
  Using cached https://www.piwheels.org/simple/werkzeug/Werkzeug-1.0.1-py2.py3-none-any.whl (298 kB)
Collecting google-auth<2,>=1.6.3
  Using cached google_auth-1.27.1-py2.py3-none-any.whl (136 kB)
Collecting google-auth-oauthlib<0.5,>=0.4.1
  Using cached google_auth_oauthlib-0.4.3-py2.py3-none-any.whl (18 kB)
Collecting markdown>=2.6.8
  Using cached https://www.piwheels.org/simple/markdown/Markdown-3.3.4-py3-none-any.whl (97 kB)
Collecting requests<3,>=2.21.0
  Downloading https://www.piwheels.org/simple/requests/requests-2.25.1-py2.py3-none-any.whl (61 kB)
     |████████████████████████████████| 61 kB 1.4 MB/s
Collecting setuptools>=41.0.0
  Using cached setuptools-54.1.0-py3-none-any.whl (784 kB)
Collecting tensorboard-plugin-wit>=1.6.0
  Using cached tensorboard_plugin_wit-1.8.0-py3-none-any.whl (781 kB)
Collecting cachetools<5.0,>=2.0.0
  Using cached https://www.piwheels.org/simple/cachetools/cachetools-4.2.1-py3-none-any.whl (12 kB)
Collecting pyasn1-modules>=0.2.1
  Using cached https://www.piwheels.org/simple/pyasn1-modules/pyasn1_modules-0.2.8-py2.py3-none-any.whl (155 kB)
Collecting rsa<5,>=3.1.4
  Using cached https://www.piwheels.org/simple/rsa/rsa-4.7.2-py3-none-any.whl (34 kB)
Collecting requests-oauthlib>=0.7.0
  Using cached https://www.piwheels.org/simple/requests-oauthlib/requests_oauthlib-1.3.0-py2.py3-none-any.whl (23 kB)
Collecting importlib-metadata
  Using cached https://www.piwheels.org/simple/importlib-metadata/importlib_metadata-3.7.0-py3-none-any.whl (11 kB)
Collecting pyasn1<0.5.0,>=0.4.6
  Using cached https://www.piwheels.org/simple/pyasn1/pyasn1-0.4.8-py2.py3-none-any.whl (77 kB)
Collecting certifi>=2017.4.17
  Downloading https://www.piwheels.org/simple/certifi/certifi-2020.12.5-py2.py3-none-any.whl (147 kB)
     |████████████████████████████████| 147 kB 2.5 MB/s
Collecting urllib3<1.27,>=1.21.1
  Downloading https://www.piwheels.org/simple/urllib3/urllib3-1.26.3-py2.py3-none-any.whl (137 kB)
     |████████████████████████████████| 137 kB 3.5 MB/s
Collecting idna<3,>=2.5
  Downloading https://www.piwheels.org/simple/idna/idna-2.10-py2.py3-none-any.whl (58 kB)
     |████████████████████████████████| 58 kB 1.9 MB/s
Collecting chardet<5,>=3.0.2
  Downloading https://www.piwheels.org/simple/chardet/chardet-4.0.0-py2.py3-none-any.whl (178 kB)
     |████████████████████████████████| 178 kB 4.3 MB/s
Collecting oauthlib>=3.0.0
  Using cached https://www.piwheels.org/simple/oauthlib/oauthlib-3.1.0-py2.py3-none-any.whl (147 kB)
Collecting zipp>=0.5
  Using cached zipp-3.4.1-py3-none-any.whl (5.2 kB)
Installing collected packages: urllib3, pyasn1, idna, chardet, certifi, zipp, typing-extensions, six, setuptools, rsa, requests, pyasn1-modules, oauthlib, cachetools, requests-oauthlib, importlib-metadata, google-auth, wheel, werkzeug, tensorboard-plugin-wit, protobuf, numpy, markdown, grpcio, google-auth-oauthlib, absl-py, wrapt, termcolor, tensorflow-estimator, tensorboard, opt-einsum, keras-preprocessing, h5py, google-pasta, gast, flatbuffers, astunparse, tensorflow
  Attempting uninstall: urllib3
    Found existing installation: urllib3 1.24.1
    Not uninstalling urllib3 at /usr/lib/python3/dist-packages, outside environment /home/pi/google-kws/venv
    Can't uninstall 'urllib3'. No files were found to uninstall.
  Attempting uninstall: pyasn1
    Found existing installation: pyasn1 0.4.8
    Uninstalling pyasn1-0.4.8:
      Successfully uninstalled pyasn1-0.4.8
  Attempting uninstall: idna
    Found existing installation: idna 2.6
    Not uninstalling idna at /usr/lib/python3/dist-packages, outside environment /home/pi/google-kws/venv
    Can't uninstall 'idna'. No files were found to uninstall.
  Attempting uninstall: chardet
    Found existing installation: chardet 3.0.4
    Not uninstalling chardet at /usr/lib/python3/dist-packages, outside environment /home/pi/google-kws/venv
    Can't uninstall 'chardet'. No files were found to uninstall.
  Attempting uninstall: certifi
    Found existing installation: certifi 2018.8.24
    Not uninstalling certifi at /usr/lib/python3/dist-packages, outside environment /home/pi/google-kws/venv
    Can't uninstall 'certifi'. No files were found to uninstall.
  Attempting uninstall: zipp
    Found existing installation: zipp 3.4.1
    Uninstalling zipp-3.4.1:
      Successfully uninstalled zipp-3.4.1
  Attempting uninstall: typing-extensions
    Found existing installation: typing-extensions 3.7.4.3
    Uninstalling typing-extensions-3.7.4.3:
      Successfully uninstalled typing-extensions-3.7.4.3
  Attempting uninstall: six
    Found existing installation: six 1.15.0
    Uninstalling six-1.15.0:
      Successfully uninstalled six-1.15.0
  Attempting uninstall: setuptools
    Found existing installation: setuptools 54.1.0
    Uninstalling setuptools-54.1.0:
      Successfully uninstalled setuptools-54.1.0
  Attempting uninstall: rsa
    Found existing installation: rsa 4.7.2
    Uninstalling rsa-4.7.2:
      Successfully uninstalled rsa-4.7.2
  Attempting uninstall: requests
    Found existing installation: requests 2.21.0
    Not uninstalling requests at /usr/lib/python3/dist-packages, outside environment /home/pi/google-kws/venv
    Can't uninstall 'requests'. No files were found to uninstall.
  Attempting uninstall: pyasn1-modules
    Found existing installation: pyasn1-modules 0.2.8
    Uninstalling pyasn1-modules-0.2.8:
      Successfully uninstalled pyasn1-modules-0.2.8
  Attempting uninstall: oauthlib
    Found existing installation: oauthlib 3.1.0
    Uninstalling oauthlib-3.1.0:
      Successfully uninstalled oauthlib-3.1.0
  Attempting uninstall: cachetools
    Found existing installation: cachetools 4.2.1
    Uninstalling cachetools-4.2.1:
      Successfully uninstalled cachetools-4.2.1
  Attempting uninstall: requests-oauthlib
    Found existing installation: requests-oauthlib 1.3.0
    Uninstalling requests-oauthlib-1.3.0:
      Successfully uninstalled requests-oauthlib-1.3.0
  Attempting uninstall: importlib-metadata
    Found existing installation: importlib-metadata 3.7.0
    Uninstalling importlib-metadata-3.7.0:
      Successfully uninstalled importlib-metadata-3.7.0
  Attempting uninstall: google-auth
    Found existing installation: google-auth 1.27.1
    Uninstalling google-auth-1.27.1:
      Successfully uninstalled google-auth-1.27.1
  Attempting uninstall: wheel
    Found existing installation: wheel 0.36.2
    Uninstalling wheel-0.36.2:
      Successfully uninstalled wheel-0.36.2
  Attempting uninstall: werkzeug
    Found existing installation: Werkzeug 1.0.1
    Uninstalling Werkzeug-1.0.1:
      Successfully uninstalled Werkzeug-1.0.1
  Attempting uninstall: tensorboard-plugin-wit
    Found existing installation: tensorboard-plugin-wit 1.8.0
    Uninstalling tensorboard-plugin-wit-1.8.0:
      Successfully uninstalled tensorboard-plugin-wit-1.8.0
  Attempting uninstall: protobuf
    Found existing installation: protobuf 3.15.5
    Uninstalling protobuf-3.15.5:
      Successfully uninstalled protobuf-3.15.5
  Attempting uninstall: numpy
    Found existing installation: numpy 1.19.5
    Uninstalling numpy-1.19.5:
      Successfully uninstalled numpy-1.19.5
  Attempting uninstall: markdown
    Found existing installation: Markdown 3.3.4
    Uninstalling Markdown-3.3.4:
      Successfully uninstalled Markdown-3.3.4
  Attempting uninstall: grpcio
    Found existing installation: grpcio 1.32.0
    Uninstalling grpcio-1.32.0:
      Successfully uninstalled grpcio-1.32.0
  Attempting uninstall: google-auth-oauthlib
    Found existing installation: google-auth-oauthlib 0.4.3
    Uninstalling google-auth-oauthlib-0.4.3:
      Successfully uninstalled google-auth-oauthlib-0.4.3
  Attempting uninstall: absl-py
    Found existing installation: absl-py 0.11.0
    Uninstalling absl-py-0.11.0:
      Successfully uninstalled absl-py-0.11.0
  Attempting uninstall: wrapt
    Found existing installation: wrapt 1.12.1
    Uninstalling wrapt-1.12.1:
      Successfully uninstalled wrapt-1.12.1
  Attempting uninstall: termcolor
    Found existing installation: termcolor 1.1.0
    Uninstalling termcolor-1.1.0:
      Successfully uninstalled termcolor-1.1.0
  Attempting uninstall: tensorflow-estimator
    Found existing installation: tensorflow-estimator 2.4.0
    Uninstalling tensorflow-estimator-2.4.0:
      Successfully uninstalled tensorflow-estimator-2.4.0
  Attempting uninstall: tensorboard
    Found existing installation: tensorboard 2.4.1
    Uninstalling tensorboard-2.4.1:
      Successfully uninstalled tensorboard-2.4.1
  Attempting uninstall: opt-einsum
    Found existing installation: opt-einsum 3.3.0
    Uninstalling opt-einsum-3.3.0:
      Successfully uninstalled opt-einsum-3.3.0
  Attempting uninstall: keras-preprocessing
    Found existing installation: Keras-Preprocessing 1.1.2
    Uninstalling Keras-Preprocessing-1.1.2:
      Successfully uninstalled Keras-Preprocessing-1.1.2
  Attempting uninstall: h5py
    Found existing installation: h5py 2.10.0
    Uninstalling h5py-2.10.0:
      Successfully uninstalled h5py-2.10.0
  Attempting uninstall: google-pasta
    Found existing installation: google-pasta 0.2.0
    Uninstalling google-pasta-0.2.0:
      Successfully uninstalled google-pasta-0.2.0
  Attempting uninstall: gast
    Found existing installation: gast 0.3.3
    Uninstalling gast-0.3.3:
      Successfully uninstalled gast-0.3.3
  Attempting uninstall: flatbuffers
    Found existing installation: flatbuffers 1.12
    Uninstalling flatbuffers-1.12:
      Successfully uninstalled flatbuffers-1.12
  Attempting uninstall: astunparse
    Found existing installation: astunparse 1.6.3
    Uninstalling astunparse-1.6.3:
      Successfully uninstalled astunparse-1.6.3
  Attempting uninstall: tensorflow
    Found existing installation: tensorflow 2.4.1
    Uninstalling tensorflow-2.4.1:
      Successfully uninstalled tensorflow-2.4.1
Successfully installed absl-py-0.11.0 astunparse-1.6.3 cachetools-4.2.1 certifi-2020.12.5 chardet-4.0.0 flatbuffers-1.12 gast-0.3.3 google-auth-1.27.1 google-auth-oauthlib-0.4.3 google-pasta-0.2.0 grpcio-1.32.0 h5py-2.10.0 idna-2.10 importlib-metadata-3.7.0 keras-preprocessing-1.1.2 markdown-3.3.4 numpy-1.19.5 oauthlib-3.1.0 opt-einsum-3.3.0 protobuf-3.15.5 pyasn1-0.4.8 pyasn1-modules-0.2.8 requests-2.25.1 requests-oauthlib-1.3.0 rsa-4.7.2 setuptools-54.1.0 six-1.15.0 tensorboard-2.4.1 tensorboard-plugin-wit-1.8.0 tensorflow-2.4.1 tensorflow-estimator-2.4.0 termcolor-1.1.0 typing-extensions-3.7.4.3 urllib3-1.26.3 werkzeug-1.0.1 wheel-0.36.2 wrapt-1.12.1 zipp-3.4.1
```

**Comment by Qengineering** [comment]
This error hints at two different versions of numpy on your machine. (virtual environment?) Or a incompatible numpy version.
[Link](https://github.com/tensorflow/tensorflow/issues/44654).
Given the message it can also be h5py generating the error.

**Comment by Qengineering** [comment]
Just installed TF 2.4.1 on a fesh RPi 64-bit OS. It turns out that the TF wheel has replaced my initial numpy version 1.20.1 for an 'old'  version 1.19.5. This is how it should work. For now, TF doesn't work with Numpy 1.20.

**Comment by StuartIanNaylor** [comment]
Yeah its a vanilla install of PiOS 64 & Tensorflow all installed in a single virtual environment.
I was really happy to see numpy 1.19.5 as my own failed compile used 20.x and thought yeah! but no, if TF has the wrong version like mine it bangs at much earlier with a simpler msg.
Because I did a --upgrade with a force reinstall then are not all the packages what is dictated by tensorflow-2.4.1 and its not possible to have a simpler install?
Thinking your right as  h5py is also compiled with numpy and guess you have numpy correct but the latest version of h5py might be using 20.x? But has been compiled in with h5py==2.10.0 ( but that been used for some time) so sort of confused as h5py==2.1.0 uses numpy 1.7.0 so thinking I need a version of h5py that uses numpy 1.20.x as maybe it wa something else wrong with my compile

**Comment by StuartIanNaylor** [comment]
I think problem is just installing h5py 1st as Requirement already satisfied: numpy>=1.7 so its pulls in a compiles with the latest.
I pulled in numpy 1.19.5 1st so h5py compiles with it then I will install tensorflow & h5py with numpy 1.19.5 which seems to work so now I am back with my battle with tensorflow-addons.

PS do you have a .bazelrc  from the tensorflow build ?

**Comment by Qengineering** [comment]
Tip. Use the next  line in configure.py from [koens](https://github.com/koenvervloesem/tensorflow-addons-on-arm) page.
```
def is_raspi_arm():
    return (os.uname()[4] == "armv7l") or (os.uname()[4] == "aarch64")  
```

**Comment by Qengineering** [comment]
Tip 2. Remove the -mavx flag
```
    if is_macos() or is_linux():
#        write("build --copt=-mavx") <-not known in gcc aarch64 
        write("build --cxxopt=-std=c++14")
        write("build --host_cxxopt=-std=c++14")
```

**Comment by StuartIanNaylor** [comment]
Yeah I have been just manually editing the bazelrc after to similar effect but really should do the above.

I started due to the really great models of kws that work well on x86_64 https://github.com/google-research/google-research/tree/master/kws_streaming

I have another problem now

```
2021-03-06 14:07:51.057994: W tensorflow/c/c_api.cc:326] Operation '{name:'gru/cell/while' id:665 op device:{} def:{{{node gru/cell/while}} = While[T=[DT_INT32, DT_INT32, DT_INT32, DT_VARIANT, DT_FLOAT, ..., DT_VARIANT, DT_VARIANT, DT_VARIANT, DT_VARIANT, DT_VARIANT], _lower_using_switch_merge=true, _num_original_outputs=52, _read_only_resource_inputs=[7, 8, 9], body=gru_cell_while_body_1257[], cond=gru_cell_while_cond_1256[], output_shapes=[[], [], [], [], [100,256], ..., [], [], [], [], []], parallel_iterations=32](gru/cell/while/loop_counter, gru/cell/while/maximum_iterations, gru/cell/time, gru/cell/TensorArrayV2_1, gru/cell/ReadVariableOp, gru/cell/strided_slice, gru/cell/TensorArrayUnstack/TensorListFromTensor, gru/cell/gru_cell/bias, gru/cell/gru_cell/kernel, gru/cell/gru_cell/recurrent_kernel, gru/cell/while/EmptyTensorList, gru/cell/while/EmptyTensorList_1, gru/cell/while/EmptyTensorList_2, gru/cell/while/EmptyTensorList_3, gru/cell/while/EmptyTensorList_4, gru/cell/while/EmptyTensorList_5, gru/cell/while/EmptyTensorList_6, gru/cell/while/EmptyTensorList_7, gru/cell/while/EmptyTensorList_8, gru/cell/while/EmptyTensorList_9, gru/cell/while/EmptyTensorList_10, gru/cell/while/EmptyTensorList_11, gru/cell/while/EmptyTensorList_12, gru/cell/while/EmptyTensorList_13, gru/cell/while/EmptyTensorList_14, gru/cell/while/EmptyTensorList_15, gru/cell/while/EmptyTensorList_16, gru/cell/while/EmptyTensorList_17, gru/cell/while/EmptyTensorList_18, gru/cell/while/EmptyTensorList_19, gru/cell/while/EmptyTensorList_20, gru/cell/while/EmptyTensorList_21, gru/cell/while/EmptyTensorList_22, gru/cell/while/EmptyTensorList_23, gru/cell/while/EmptyTensorList_24, gru/cell/while/EmptyTensorList_25, gru/cell/while/EmptyTensorList_26, gru/cell/while/EmptyTensorList_27, gru/cell/while/EmptyTensorList_28, gru/cell/while/EmptyTensorList_29, gru/cell/while/EmptyTensorList_30, gru/cell/while/EmptyTensorList_31, gru/cell/while/EmptyTensorList_32, gru/cell/while/EmptyTensorList_33, gru/cell/while/EmptyTensorList_34, gru/cell/while/EmptyTensorList_35, gru/cell/while/EmptyTensorList_36, gru/cell/while/EmptyTensorList_37, gru/cell/while/EmptyTensorList_38, gru/cell/while/EmptyTensorList_39, gru/cell/while/EmptyTensorList_40, gru/cell/while/EmptyTensorList_41)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.
```
Please use tf.compat.v1.keras.backend.set_session instead... What is really annoying is at one time I had it working in full but would crash on export at end of training but the GRU and custom ops worked fine.
I went off trying various other tensorflows and installs and now can not remember how I got it to work.
I am not sure why I am getting the above messages as with the X86_64 those messages don't show

**Comment by StuartIanNaylor** [comment]
@Qengineering  Hey many thanks as went back to the x86_64 version and Doh! it does do exactly the same.
I can tell the difference though as the model on Aarch64 never seems to get accurate whilst on X86-64 it garners quickly.

Anyway just added 
```        
        write("build --cxxopt=-std=c++14")
        write("build --host_cxxopt=-std=c++14")
```

Instaly accuracy is garnered early and now have to see if it still crashes on export but thinks for the input as much appreciated as was going to give up on it.
It may of been me also as just found that if you create a too big a training rate the model just seems to get confused and stays fairly random on accuracy.

**Comment by Qengineering** [comment]
Great to hear!
Thanks for keeping me posted!

**Comment by masip85** [comment]
> Just installed TF 2.4.1 on a fesh RPi 64-bit OS. It turns out that the TF wheel has replaced my initial numpy version 1.20.1 for an 'old' version 1.19.5. Perhaps this could also trigger your error. Maybe a re-install of numpy may fix it? `sudo -H pip3 install -U numpy`

Same error for me, but if I update numpy to 1.20.0, then I get this:
```
Python 3.8.10 | packaged by conda-forge | (default, May 11 2021, 06:25:29) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import tensorflow as tf
/opt/conda/lib/python3.8/site-packages/h5py/__init__.py:37: UserWarning: h5py is running against HDF5 1.10.6 when it was built against 1.10.4, this may cause problems
  _warn(("h5py is running against HDF5 {0} when it was built against {1}, "
Warning! ***HDF5 library version mismatched error***
The HDF5 header files used to compile this application do not match
the version used by the HDF5 library to which this application is linked.
Data corruption or segmentation faults may occur if the application continues.
This can happen when an application was compiled by one version of HDF5 but
linked with a different version of static or shared HDF5 library.
You should recompile the application or check your shared library related
settings such as 'LD_LIBRARY_PATH'.
You can, at your own risk, disable this warning by setting the environment
variable 'HDF5_DISABLE_VERSION_CHECK' to a value of '1'.
Setting it to 2 or higher will suppress the warning messages totally.
Headers are 1.10.4, library is 1.10.6
            SUMMARY OF THE HDF5 CONFIGURATION
            =================================

General Information:
-------------------
                   HDF5 Version: 1.10.6
                  Configured on: Tue Nov 24 01:35:54 UTC 2020
                  Configured by: conda@b961255d19ea
                    Host system: aarch64-conda-linux-gnu
              Uname information: Linux b961255d19ea 4.15.0-36-generic #39-Ubuntu SMP Wed Sep 26 06:05:44 UTC 2018 aarch64 aarch64 aarch64 GNU/Linux
                       Byte sex: little-endian
             Installation point: /drone/src/build_artifacts/hdf5_split_1606181292220/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh

Compiling Options:
------------------
                     Build Mode: production
              Debugging Symbols: no
                        Asserts: no
                      Profiling: no
             Optimization Level: high

Linking Options:
----------------
                      Libraries: static, shared
  Statically Linked Executables: 
                        LDFLAGS: -Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,-rpath,/drone/src/build_artifacts/hdf5_split_1606181292220/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib -Wl,-rpath-link,/drone/src/build_artifacts/hdf5_split_1606181292220/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib -L/drone/src/build_artifacts/hdf5_split_1606181292220/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib
                     H5_LDFLAGS: 
                     AM_LDFLAGS:  -L/drone/src/build_artifacts/hdf5_split_1606181292220/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib
                Extra libraries: -lcrypto -lcurl -lpthread -lz -ldl -lm 
                       Archiver: /drone/src/build_artifacts/hdf5_split_1606181292220/_build_env/bin/aarch64-conda-linux-gnu-ar
                       AR_FLAGS: cr
                         Ranlib: /drone/src/build_artifacts/hdf5_split_1606181292220/_build_env/bin/aarch64-conda-linux-gnu-ranlib

Languages:
----------
                              C: yes
                     C Compiler: /drone/src/build_artifacts/hdf5_split_1606181292220/_build_env/bin/aarch64-conda-linux-gnu-cc
                       CPPFLAGS: -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem /drone/src/build_artifacts/hdf5_split_1606181292220/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/include
                    H5_CPPFLAGS: -D_GNU_SOURCE -D_POSIX_C_SOURCE=200809L   -DNDEBUG -UH5_DEBUG_API
                    AM_CPPFLAGS:  -I/drone/src/build_artifacts/hdf5_split_1606181292220/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/include
                        C Flags: -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O3 -pipe -isystem /drone/src/build_artifacts/hdf5_split_1606181292220/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/include -fdebug-prefix-map=/drone/src/build_artifacts/hdf5_split_1606181292220/work=/usr/local/src/conda/hdf5_split-1.10.6 -fdebug-prefix-map=/drone/src/build_artifacts/hdf5_split_1606181292220/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh=/usr/local/src/conda-prefix
                     H5 C Flags:  -std=c99  -pedantic -Wall -Wextra -Wbad-function-cast -Wc++-compat -Wcast-align -Wcast-qual -Wconversion -Wdeclaration-after-statement -Wdisabled-optimization -Wfloat-equal -Wformat=2 -Wno-format-nonliteral -Winit-self -Winvalid-pch -Wmissing-declarations -Wmissing-include-dirs -Wmissing-prototypes -Wnested-externs -Wold-style-definition -Wpacked -Wredundant-decls -Wshadow -Wstrict-prototypes -Wswitch-enum -Wswitch-default -Wundef -Wunused-macros -Wunsafe-loop-optimizations -Wwrite-strings -Wlogical-op -Wlarger-than=2560 -Wsync-nand -Wframe-larger-than=16384 -Wpacked-bitfield-compat -Wstrict-overflow=5 -Wjump-misses-init -Wunsuffixed-float-constants -Wdouble-promotion -Wtrampolines -Wstack-usage=8192 -Wvector-operation-performance -Wdate-time -Warray-bounds=2 -Wc99-c11-compat -Wnull-dereference -Wunused-const-variable -Wduplicated-cond -Whsa -Wnormalized -Walloc-zero -Walloca -Wduplicated-branches -Wformat-overflow=2 -Wformat-truncation=2 -Wimplicit-fallthrough=5 -Wrestrict -fstdarg-opt -s -Wno-inline -Wno-aggregate-return -Wno-missing-format-attribute -Wno-missing-noreturn -Wno-suggest-attribute=const -Wno-suggest-attribute=pure -Wno-suggest-attribute=noreturn -Wno-suggest-attribute=format -O3
                     AM C Flags: 
               Shared C Library: yes
               Static C Library: yes


                        Fortran: yes
               Fortran Compiler: /drone/src/build_artifacts/hdf5_split_1606181292220/_build_env/bin/aarch64-conda-linux-gnu-gfortran
                  Fortran Flags: 
               H5 Fortran Flags:  -pedantic -Wall -Wextra -Wunderflow -Wimplicit-interface -Wsurprising -Wno-c-binding-type  -s -O2
               AM Fortran Flags: 
         Shared Fortran Library: yes
         Static Fortran Library: yes

                            C++: yes
                   C++ Compiler: /drone/src/build_artifacts/hdf5_split_1606181292220/_build_env/bin/aarch64-conda-linux-gnu-c++
                      C++ Flags: -fvisibility-inlines-hidden -std=c++17 -fmessage-length=0 -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O3 -pipe -isystem /drone/src/build_artifacts/hdf5_split_1606181292220/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/include -fdebug-prefix-map=/drone/src/build_artifacts/hdf5_split_1606181292220/work=/usr/local/src/conda/hdf5_split-1.10.6 -fdebug-prefix-map=/drone/src/build_artifacts/hdf5_split_1606181292220/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh=/usr/local/src/conda-prefix
                   H5 C++ Flags:   -pedantic -Wall -W -Wundef -Wshadow -Wpointer-arith -Wcast-qual -Wcast-align -Wwrite-strings -Wconversion -Wredundant-decls -Winline -Wsign-promo -Woverloaded-virtual -Wold-style-cast -Weffc++ -Wreorder -Wnon-virtual-dtor -Wctor-dtor-privacy -Wabi -finline-functions -s -O
                   AM C++ Flags: 
             Shared C++ Library: yes
             Static C++ Library: yes

                           Java: no


Features:
---------
                   Parallel HDF5: no
Parallel Filtered Dataset Writes: no
              Large Parallel I/O: no
              High-level library: yes
                Build HDF5 Tests: yes
                Build HDF5 Tools: yes
                    Threadsafety: yes
             Default API mapping: v110
  With deprecated public symbols: yes
          I/O filters (external): deflate(zlib)
                             MPE: no
                      Direct VFD: no
              (Read-Only) S3 VFD: yes
            (Read-Only) HDFS VFD: no
                         dmalloc: no
  Packages w/ extra debug output: none
                     API tracing: no
            Using memory checker: yes
 Memory allocation sanity checks: no
          Function stack tracing: no
       Strict file format checks: no
    Optimization instrumentation: no
Bye...
Aborted (core dumped)
```

**Comment by Qengineering** [comment]
Dear @masip85,

There are several issues with Numpy 1.20 and TF see [link](https://github.com/tensorflow/tensorflow/issues/47691). As you can see [here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/pip_package/setup.py), TF's dependency is linked to version 1.19 of Numpy. Given the efforts of porting TF to version 1.20, it will be available soon, perhaps TF 2.5.0?
For now, stick to Numpy 1.19 and make sure other software, like h5py, is compiled with Numpy 1.19.
